src/cli/lurk_proof.rs

```diff
@@ -0,0 +1,68 @@
+use serde::{Deserialize, Serialize};
+
+use anyhow::Result;
+
+use lurk::{
+    eval::lang::{Coproc, Lang},
+    proof::{nova, nova::PublicParams},
+};
+
+type F = pasta_curves::pallas::Scalar;
+
+#[derive(Serialize, Deserialize)]
+pub struct NovaProof<'a> {
+    pub proof: nova::Proof<'a, Coproc<F>>,
+    pub public_inputs: Vec<F>,
+    pub public_outputs: Vec<F>,
+    pub num_steps: usize,
+}
+
+impl<'a> NovaProof<'a> {
+    pub(crate) fn verify(&self, pp: &PublicParams<'_, Coproc<F>>) -> Result<bool> {
+        let Self {
+            proof,
+            public_inputs,
+            public_outputs,
+            num_steps,
+        } = self;
+        Ok(proof.verify(pp, *num_steps, public_inputs, public_outputs)?)
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub enum LurkProofMeta {
+    Evaluation {
+        input: String,
+        environment: String,
+        output: String,
+    },
+    Opening {
+        input: String,
+        output: String,
+    },
+}
```

Comment:
These strings may need to be optional, since in the general case there may not be a readable string representation for the input or environment.

Reply:
What's the criteria? How would the user choose?

Reply:
I don't understand. It's possible to have a proof of an expression that contains unreadable values (like a `Thunk`, for example). It's not a matter of choice, it's a matter of necessity — if the intention is to be general here.

Reply:
I guess the meta itself could be entirely optional. But if having those strings is required, and they have to be correct, then this will make some valid proofs impossible to represent.

Reply:
Done